src/main/java/de/tum/in/www1/artemis/service/connectors/apollon/ApollonConversionService.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.service.connectors.apollon;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.service.connectors.apollon.dto.ApollonModelDTO;
+
+@Service
+@Profile("apollon")
+public class ApollonConversionService {
+
+    private final Logger log = LoggerFactory.getLogger(ApollonConversionService.class);
+
+    @Value("${artemis.apollon.conversion-service-url}")
+    private String apollonConversionUrl;
+
+    private RestTemplate restTemplate;
+
+    public ApollonConversionService(RestTemplate apollonRestTemplate) {
+        setRestTemplate(apollonRestTemplate);
```

Comment:
Do you really need the setter? Can't you just `this.restTemplate = appollonRestTemplate`?

Reply:
It is needed for testing and does not alter the behavior here